services/ec2/src/main/java/org/finra/gatekeeper/services/accessrequest/delegates/GrantAccessServiceTask.java

```diff
@@ -86,44 +86,51 @@ public void execute(DelegateExecution execution) throws Exception {
         AccessRequest accessRequest = accessRequestService.updateInstanceStatus((AccessRequest) execution.getVariable("accessRequest"));
         logger.info("Granting Access to " + accessRequest);
         try {
-
-            // Prepare parameters
-            AWSEnvironment env = new AWSEnvironment(accessRequest.getAccount(), accessRequest.getRegion());
-            logger.info("Environment for this access request is " + env.getAccount() + " ( " + env.getRegion() + " )");
-            List<String> instances = accessRequest.getInstances()
-                    .stream().filter(instance -> instance.getStatus().equals("Online"))
-                    .map(AWSInstance::getInstanceId)
-                    .collect(Collectors.toList());
-
-            // Do all of this for each user in the request
-            if(instances.size()>0) {
-                List<GatekeeperLinuxNotification> linuxNotifications = new ArrayList<>();
-                List<GatekeeperWindowsNotification> windowsNotifications = new ArrayList<>();
-
-                String platform = accessRequest.getPlatform();
-                switch(platform){
-                    case "Linux":
-                        linuxNotifications = createLinuxUser(accessRequest, env, instances, platform);
-                        break;
-                    case "Windows":
-                        windowsNotifications = createWindowsUser(accessRequest, env, instances, platform);
-                    break;
-                    default:
-                        throw new GatekeeperException("Unsupported platform " + platform);
-                }
-
-                // If an SNS topic is provided run the queries, otherwise lets skip this step.
+            //Handle No User Creation
+            if(accessRequest.isNoUser() != null && accessRequest.isNoUser()){
+                logger.info("No users were requested with this request. Pushing to SNS Topic.");
                 if(snsService.isTopicSet()) {
                     snsService.pushToSNSTopic(accessRequestService.getLiveRequestsForUsersInRequest(EventType.APPROVAL, accessRequest));
                 } else {
                     logger.info("Skip querying of live request data as SNS topic ARN was not provided");
                 }
```

Comment:
If I understand this correctly through the diff, this sns logic is the same as below? why not if statement around the SSM bit and just have it execute the already existing SNS block? Or even helper function this block?

Reply:
![Screen Shot 2023-05-26 at 10 02 51 AM](https://github.com/FINRAOS/Gatekeeper/assets/5823989/ad0c51ec-b7f2-4a99-a15c-95180da36163)

Reply:
Those need to be separate. When creating users we don't want to also push to the sns if user creation fails. I've added as a helper function